projects/extension/sql/idempotent/905-text-to-sql.sql

```diff
@@ -0,0 +1,510 @@
+        , description pg_catalog.text
+        )
+        where f.objtype operator(pg_catalog.=) 'function'
+    ) x
+    ;
+
+    -- find relevant sql examples
+    select pg_catalog.string_agg
+    ( pg_catalog.format
+      ( E'<example-sql>\n/*\n%s\n*/\n%s\n</example-sql>'
+      , r.description
+      , r.sql
+      )
+    , E'\n\n'
+    ) into _relevant_sql
+    from ai._find_relevant_sql
+    ( _catalog_id
+    , _prompt_emb
+    , "limit"=>"limit"
+    , max_dist=>max_dist
+    ) r
+    ;
+
+    -- construct overall prompt
+    select pg_catalog.concat_ws
+    ( E'\n'
+    , 'Consider the following context when responding.'
+    , 'Any relevant table, view, and functions descriptions and DDL definitions will appear in <table></table>, <view></view>, and <function></function> tags respectively.'
+    , 'Any relevant example SQL statements will appear in <example-sql></example-sql> tags.'
+    , _tbl_ctx
+    , _view_ctx
+    , _func_ctx
+    , _relevant_sql
+    , 'Respond to the following question with a SQL statement only. Only use syntax and functions that work with PostgreSQL.'
+    , 'Q: ' operator(pg_catalog.||) prompt
+    , 'A: '
+    ) into strict _prompt
+    ;
+
+    return _prompt;
+end
+$func$ language plpgsql stable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- text_to_sql_openai
+create or replace function ai.text_to_sql_openai
+( model pg_catalog.text
+, api_key pg_catalog.text default null
+, api_key_name pg_catalog.text default null
+, base_url pg_catalog.text default null
+, frequency_penalty pg_catalog.float8 default null
+, logit_bias pg_catalog.jsonb default null
+, logprobs pg_catalog.bool default null
+, top_logprobs pg_catalog.int4 default null
+, max_tokens pg_catalog.int4 default null
+, n pg_catalog.int4 default null
+, presence_penalty pg_catalog.float8 default null
+, seed pg_catalog.int4 default null
+, stop pg_catalog.text default null
+, temperature pg_catalog.float8 default null
+, top_p pg_catalog.float8 default null
+, openai_user pg_catalog.text default null
+) returns pg_catalog.jsonb
+as $func$
+    select json_object
+    ( 'provider': 'openai'
+    , 'model': model
+    , 'api_key': api_key
+    , 'api_key_name': api_key_name
+    , 'base_url': base_url
+    , 'frequency_penalty': frequency_penalty
+    , 'logit_bias': logit_bias
+    , 'logprobs': logprobs
+    , 'top_logprobs': top_logprobs
+    , 'max_tokens': max_tokens
+    , 'n': n
+    , 'presence_penalty': presence_penalty
+    , 'seed': seed
+    , 'stop': stop
+    , 'temperature': temperature
+    , 'top_p': top_p
+    , 'openai_user': openai_user
+    absent on null
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- text_to_sql_ollama
+create or replace function ai.text_to_sql_ollama
```

Comment:
append _config to the names?
```suggestion
create or replace function ai.text_to_sql_ollama_config
```